valhalla/baldr/time_info.h

```diff
@@ -0,0 +1,226 @@
+#pragma once
+
+#include <chrono>
+
+#include <valhalla/baldr/datetime.h>
+#include <valhalla/baldr/graphid.h>
+#include <valhalla/baldr/graphreader.h>
+#include <valhalla/midgard/logging.h>
+#include <valhalla/proto/api.pb.h>
+
+namespace dt = valhalla::baldr::DateTime;
+namespace sc = std::chrono;
+
+namespace valhalla {
+namespace baldr {
+// A structure for tracking time information as the route progresses
+struct TimeInfo {
```

Comment:
the main point of this struct is to provide a one stop shop for 2 things:

1. convenient encapsulation that allows any routing algorithm (graph expansion) to track time
2. parameter pack that we can, in the future, send directly to costing to do the right thign with respect to time-based speed look up (second of week for historical speed and seconds from now for real time speed)